text/0000-abort-by-default.md

```diff
@@ -0,0 +1,70 @@
+- Feature Name: abort_by_default
+- Start Date: 2016-10-01
+- RFC PR: (leave this empty)
+- Rust Issue: (leave this empty)
+
+# Summary
+[summary]: #summary
+
+Use abort as the standard panic method rather than unwind.
+
+# Motivation
+[motivation]: #motivation
+
+## Performance
+
+Generally, the performance of bigger programs [improves by 10%](https://www.youtube.com/watch?v=mRGb4hoGuPs), which is no small amount. When overflow checking is enabled, it is as high as 2x, making overflowing checking plausible in production.
+
+The performance gains mainly originates in the fact that it is a lot easier for the compiler to optimize, due to the unwind path disappearing, and hence reasoning about the code becomes easier.
+
+## Binary size
+
+Binary size improves by 10% as well. This is due to the lack of landing pads which are injected in the stack when unwinding is enabled.
+
+## Compile time
+
+Compile time in debug mode improves by 20% on average. In release mode, the number is around 10%.
+
+## Runtime size
+
+Unwinding requires a fair amount of runtime code, which can be seen as conflicting with the goals of Rust.
+
+## Correctness
+
+You often see people abusing `std::panic::catch_unwind` for exception handling. Forcing the user to explicitly opt in to unwinding will make it harder to unintentionally misuse it.
+
+# Detailed design
+[design]: #detailed-design
+
+Default to `panic=abort`.
+
+## Backwards compatibility
+
+No code should rely on a particular default panicking strategy, and code which do is already broken. However, this broken code might start actually encountering the issues. In a sense, that is actually positive, since it reveals that the code is broken.
```

Comment:
panics aren't promised to be unwinds, and the docs for catch_unwind call this out explicitly. The argument is that code that relies on panic=unwind for normal function is already broken because it can't guarantee the panic strategy. That said, there are some useful things (for testing, primarily) that do need to ensure panic=unwind, which is why folks elsewhere in this thread are looking for a way to support them.